packages/core/src/utils/check.ts

```diff
@@ -57,6 +65,17 @@ export function isPlainObject(value: unknown): value is object {
   return prototype === null || prototype === Object.prototype;
 }
 
+/**
+ * This function is the same as {@link isPlainObject}, but types the result as a Record / Dictionary.
+ * This function won't be necessary starting with TypeScript 4.9, thanks to improvements to the TS object type,
```

Comment:
Add an expiring TODO for it as well?

Reply:
For may 2023 or so?

Reply:
Is that when we drop TS 4.8 support already?

Reply:
Not sure
Last we discussed we'd support a TS version for 6 months, but if we start doing yearly major releases, we may be able keep supporting a TS version for the entire lifecycle of the major version & drop it in the next major

In this case that'd be v8

Reply:
That might be good to discuss on Thursday's Monthly. For now let's just leave it like this so we can merge

Reply:
I'll leave the merging up to you in case you want to add something to the commit body